### ISSUE TITLE
-Wall gives spurious "is not used" warning when expanding variadic capture and other pack

### ISSUE THREAD

**Issue by Aster89** [llvm/llvm-project#112132]
```cpp
auto foo(auto ...x) {
    return [...x = x](auto ...f) {
        (f(x), ...);
    };
}

int main() {
    foo(0)([](auto x){ return x; });
}
```
Compiling the code above via
```bash
clang++ -std=c++20 -Wall file.cpp
```
produces this spurious warning:
```none
file.cpp:2:16: warning: lambda capture 'x' is not used [-Wunused-lambda-capture]
    2 |     return [...x = x](auto ...f) {
      |                ^
file.cpp:8:5: note: in instantiation of function template specialization 'foo<int>' requested here
    8 |     foo(0)([](auto x){ return x; });
      |     ^
1 warning generated.
```
---
```
$ clang++ --version
clang version 18.1.8
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by EugeneZelenko** [comment]
Could you please try 19 or `main` branch? https://godbolt.org should be helpful.

**Comment by Aster89** [comment]
Isn't [this](https://godbolt.org/z/os9fn99TK) Clang 19.1.0?

**Comment by EugeneZelenko** [comment]
Original description tells about 18.1.8.
